src/kit/Theme/UI.tsx

```diff
@@ -1,101 +1,129 @@
-}
+import { globalCssVars, Theme, ThemeVariable } from './themeUtils';
 
-const initUI: StateUI = {
-  chromeCollapsed: false,
-  darkLight: DarkLight.Light,
-  isPageHidden: false,
-  mode: Mode.System,
-  showChrome: true,
-  showSpinner: false,
-  theme: {} as Theme,
+export { StyleProvider };
+export type { Theme };
+
+const camelCaseToKebab = (text: string): string => {
+  return text
+    .trim()
+    .split('')
+    .map((char, index) => {
+      return char === char.toUpperCase() ? `${index !== 0 ? '-' : ''}${char.toLowerCase()}` : char;
+    })
+    .join('');
 };
 
-const StoreActionUI = {
-  HideUIChrome: 'HideUIChrome',
-  HideUISpinner: 'HideUISpinner',
-  SetMode: 'SetMode',
-  SetPageVisibility: 'SetPageVisibility',
-  SetTheme: 'SetTheme',
-  ShowUIChrome: 'ShowUIChrome',
-  ShowUISpinner: 'ShowUISpinner',
-} as const;
+interface ThemeSettings {
+  className: string;
+  themeIsDark: boolean;
+  theme: Theme;
+}
 
-type ActionUI =
-  | { type: typeof StoreActionUI.HideUIChrome }
-  | { type: typeof StoreActionUI.HideUISpinner }
-  | { type: typeof StoreActionUI.SetMode; value: Mode }
-  | { type: typeof StoreActionUI.SetPageVisibility; value: boolean }
-  | { type: typeof StoreActionUI.SetTheme; value: { darkLight: DarkLight; theme: Theme } }
-  | { type: typeof StoreActionUI.ShowUIChrome }
-  | { type: typeof StoreActionUI.ShowUISpinner };
+const UIContext = React.createContext<ThemeSettings | undefined>(undefined);
 
-class UIActions {
-  constructor(private dispatch: Dispatch<ActionUI>) {}
+export const useTheme = (): { themeSettings: ThemeSettings, getThemeVar: (name: ThemeVariable) => string; } => {
+  /**
+   * Some UI Kit components such as the CodeEditor do not inherit the theme from css or page styling
+   * and instead require us to set a theme related prop dynamically. This context allows us to
+   * subscribe to UIProvider theme updates and re-render these child components with the correct
+   * theme.
+   */
+  const context = useContext(UIContext);
+  if (context === undefined) {
+    throw new Error('useStore(UI) must be used within a UIProvider');
+  }
 
-  public hideChrome = (): void => {
-    this.dispatch({ type: StoreActionUI.HideUIChrome });
-  };
+  const getThemeVar = (name: ThemeVariable): string => {
+    return context.theme[name]
+  }
 
-  public hideSpinner = (): void => {
-    this.dispatch({ type: StoreActionUI.HideUISpinner });
-  };
+  return { themeSettings: context, getThemeVar };
+};
 
-  public setMode = (mode: Mode): void => {
-    this.dispatch({ type: StoreActionUI.SetMode, value: mode });
-  };
 
-  public setPageVisibility = (isPageHidden: boolean): void => {
-    this.dispatch({ type: StoreActionUI.SetPageVisibility, value: isPageHidden });
-  };
+export const UIProvider: React.FC<{
+  children?: React.ReactNode;
+  themeIsDark?: boolean;
+  theme: Theme;
+}> = ({ children, theme, themeIsDark = false }) => {
+  const className = `ui-provider-${Math.random().toString(36).substring(2, 9)}`;
```

Comment:
this will cause the styles to be regenerated and cause all theme-sensitive components to rerender every render. instead:

* capture the `Math.random` substring as a ref (we don't want to use useMemo here because react can evict values from the memoization cache)
* hash the substring along with the `theme` and `themeIsDark` variables in the classname, and use a memo for that.